core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/hashJoin/HashJoinOpExec.scala

```diff
@@ -28,6 +28,30 @@ class HashJoinOpExec[K](
   var currentEntry: Iterator[Tuple] = _
   var currentTuple: Tuple = _
 
+  def getBuildHashTable(): ArrayBuffer[mutable.HashMap[K, ArrayBuffer[Tuple]]] = {
+    val sendingMap = new ArrayBuffer[mutable.HashMap[K, ArrayBuffer[Tuple]]]
+    var count = 1
+    var curr = new mutable.HashMap[K, ArrayBuffer[Tuple]]
+    for ((key, tuples) <- buildTableHashMap) {
+      curr.put(key, tuples._1)
+      if (count % 4000 == 0) {
```

Comment:
also why every 4000? can we give this 4000 a variable name and make it a constant or even configurable?

Reply:
This should be a constant as it is an operator thing. Different operators can have different values for this and putting it in config file is not suitable.